code/Model/SiteTreeExtension.php

```diff
@@ -75,13 +75,15 @@ public function canPublish($member)
      * before {@link SiteTree::RelativeLink()} calls {@link Controller::join_links()}
      * on the $base and $action
      *
+     * @param string &$link The URL of this page relative to siteroot including
+     * the action
      * @param string &$base The URL of this page relative to siteroot, not including
      * the action
      * @param string|boolean &$action The action or subpage called on this page.
      * (Legacy support) If this is true, then do not reduce the 'home' urlsegment
      * to an empty link
      */
-    public function updateRelativeLink(&$base, &$action)
+    public function updateRelativeLink(&$link, &$base, &$action)
```

Comment:
```suggestion
    public function updateRelativeLink(&$link, $base, $action)
```
Changes the base or action won't actually do anything (they're here for reference only) so those shouldn't be passed by reference anymore. Phpdocs for them need to be updated too.

Reply:
Well, why do we pass them at all? As you say, they don't do anything.

Reply:
As I say, for context. :p
They show what was used to generate the link in the first place and make it easier to, for example, rebuild the link using the same base but a different action (or vice versa).